packages/web/src/components/tabbedPane.tsx

```diff
@@ -36,37 +36,164 @@ export const TabbedPane: React.FunctionComponent<{
   setSelectedTab?: (tab: string) => void,
   dataTestId?: string,
   mode?: 'default' | 'select',
-}> = ({ tabs, selectedTab, setSelectedTab, leftToolbar, rightToolbar, dataTestId, mode }) => {
+  overflowMode?: 'none' | 'select'
+}> = ({ tabs, selectedTab, setSelectedTab, leftToolbar, rightToolbar, dataTestId, mode, overflowMode }) => {
   const id = React.useId();
   if (!selectedTab)
     selectedTab = tabs[0].id;
   if (!mode)
     mode = 'default';
-  return <div className='tabbed-pane' data-testid={dataTestId}>
+  if (!overflowMode)
+    overflowMode = 'none';
+
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const [visibleTabs, setVisibleTabs] = React.useState<TabbedPaneTabModel[]>(mode !== 'select' ? tabs : []);
+  const [overflowTabs, setOverflowTabs] = React.useState<TabbedPaneTabModel[]>(mode === 'select' ? tabs : []);
+  const [tabWidths, setTabWidths] = React.useState<Record<string, number>>({});
+  const [, setContainerWidth] = React.useState<number>(0);
```

Comment:
Are you using this for rerender management? Why does it matter, given you're also setting other state every time this is set? Those other sets will always be new/unique as well, so they will also cause a rerender.

Reply:
Thanks @agg23 , This makes sense. I verified, containerWidth wasn't to re-render other states. It was only used for calculation of available space. Have pushed a commit to remove it usage.